src/functionBased/components/TodoItem.js

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { FaTrash } from 'react-icons/fa';
+import styles from './TodoItem.module.css';
+
+const TodoItem = (props) => {
+  const [editing, setEditing] = useState(false);
+
+  const handleEditing = () => {
+    setEditing(true);
+  };
+
+  const handleUpdatedDone = (event) => {
+    if (event.key === 'Enter') {
+      setEditing(false);
+    }
+  };
+
+  const completedStyle = {
+    fontStyle: 'italic',
+    color: '#595959',
+    opacity: 0.4,
+    textDecoration: 'line-through',
+  };
+
+  const {
+    todo: { completed, id, title },
+    handleChangeProps,
+    deleteTodoProps,
+    setUpdate,
+  } = props;
+
+  const viewMode = {};
+  const editMode = {};
+
+  if (editing) {
+    viewMode.display = 'none';
+  } else {
+    editMode.display = 'none';
+  }
+
+  useEffect(() => () => {
+    // console.log('Cleaning up...');
+  }, []);
```

Comment:
It seems the useEffect hook here is of no use. You can kindly take it out of your code to avoid bugs, unexpected results and also to make your code cleaner, thank you